jbmc/src/java_bytecode/java_string_library_preprocess.cpp

```diff
@@ -1246,27 +1248,28 @@ java_string_library_preprocesst::get_primitive_value_of_object(
   DATA_INVARIANT(object_type.has_value(), "must have symbol for primitive");
 
   // declare tmp_type_name to hold the value
-  std::string aux_name="tmp_"+id2string(type_name);
-  symbolt symbol=get_fresh_aux_symbol(
+  const std::string aux_name = "tmp_" + id2string(type_name);
+  const symbolt symbol = get_fresh_aux_symbol(
     value_type, aux_name, aux_name, loc, ID_java, symbol_table);
-  auto value = symbol.symbol_expr();
+  const auto value = symbol.symbol_expr();
 
   // Check that the type of the object is in the symbol table,
   // otherwise there is no safe way of finding its value.
   if(
     const auto maybe_symbol =
       symbol_table.lookup(object_type->get_identifier()))
   {
-    const struct_typet &struct_type = to_struct_type(maybe_symbol->type);
+    const struct_typet struct_type = to_struct_type(maybe_symbol->type);
     // Check that the type has a value field
     const struct_union_typet::componentt &value_comp =
       struct_type.get_component("value");
     if(!value_comp.is_nil())
     {
-      pointer_typet pointer_type=::pointer_type(struct_type);
-      dereference_exprt deref(
+      const pointer_typet pointer_type = ::pointer_type(struct_type);
+      const dereference_exprt deref(
         typecast_exprt(object, pointer_type), pointer_type.subtype());
-      member_exprt deref_value(deref, value_comp.get_name(), value_comp.type());
+      const member_exprt deref_value(
```

Comment:
Use `std::move` below, don't `const`

Reply:
`code_assignt`'s constructor takes argument by const ref so `move` would have no effect here (though exprt should have constructors from movable arguments, that's outside the scope of this PR).

Reply:
`member_exprt` and `code_assignt`'s constructors take arguments by const ref so `move` would have no effect here

Reply:
True at the moment, but #3502 will eventually fix that. So whenever I have cleaned up #3502 it would be nice if code also were making use of it.